app/server/models/user.py

```diff
@@ -560,6 +560,11 @@ def check_reset_token_already_used(self, password_reset_token):
         is_valid = password_reset_token in self.password_reset_tokens
         return is_valid
 
+    def is_pin_reset_token_valid(self, pin_reset_token):
+        self.clear_expired_pin_reset_tokens()
```

Comment:
do we need to clear all remaining tokens afterwards too, I guess we don't need any once PIN is reset.

Reply:
I looked at the DB post reset and the tokens were cleared, not sure where the function for that is and I'll confirm just to be safe but it is there.